legacy-common/src/main/java/eu/arrowhead/legacy/common/model/LegacyModelConverter.java

```diff
@@ -112,12 +112,14 @@ public static ServiceRegistryRequestDTO convertLegacyServiceRegistryEntryToServi
 		final ServiceRegistryRequestDTO dto = new ServiceRegistryRequestDTO();
 		dto.setServiceDefinition(entry.getProvidedService().getServiceDefinition());
 		dto.setProviderSystem(provider);
-		dto.setServiceUri(entry.getServiceUri());
+		dto.setServiceUri(entry.getServiceURI());
 		dto.setVersion(entry.getVersion());
 		dto.setSecure(calculateSecurityType(entry.getProvidedService().getServiceMetadata()));
 		entry.getProvidedService().getServiceMetadata().remove(LegacyCommonConstants.KEY_SECURITY);
 		entry.getProvidedService().getServiceMetadata().put(LegacyCommonConstants.KEY_LEGACY_INTERFACE, entry.getProvidedService().getInterfaces().iterator().next());
-		dto.setInterfaces(List.of(LegacyCommonConstants.DEFAULT_INTERFACE));
```

Comment:
What's the reason behind having always LegacyCommonConstants.DEFAULT_INTERFACE and not the provided ones from the service itself?

I found this constant some other places, but I'm not sure the reason of it

Reply:
It's because v4.1.3 has a strict rule about how an interface has to be defined, but in v4.1.2 the interface can be any kind of string. So, we have to make sure, that a legacy client request won't be rejected due to the invalid interface definition. Therfore all the legacy service registry requests are registrated with a default interface, and the inital interface is saved into the metadata. When a orchestration request has requested interface, then the translator going to check also the metadata for the proper interface.

Reply:
Mmm assigning the original one in the metadata seems like a nice trick, but then somehow my orchestration tests weren't working because I was testing with HTTP-INSECURE-JSON. But this could also be related to the change:

.interfaces(form.getRequestedService().getInterfaces().toArray(new String[form.getRequestedService().getInterfaces().size()])) //intentionally NOT ignored

Reply:
Yes it's related to that change. Should be left as it was.